ndk-build/src/cargo.rs

```diff
@@ -14,12 +14,43 @@ pub fn cargo_ndk(
     let clang_target = format!("--target={}{}", target.ndk_llvm_triple(), sdk_version);
     let mut cargo = Command::new("cargo");
 
+    const SEP: char = '\x1f';
+
     // Read initial RUSTFLAGS
     let mut rustflags = match std::env::var("CARGO_ENCODED_RUSTFLAGS") {
         Ok(val) => val,
-        Err(std::env::VarError::NotPresent) => "".to_string(),
+        Err(std::env::VarError::NotPresent) => {
+            match std::env::var("RUSTFLAGS") {
+                Ok(val) => {
+                    // Note this is roughly equivalent to what cargo does
+                    // https://github.com/rust-lang/cargo/blob/f6de921a5d807746e972d9d10a4d8e1ca21e1b1f/src/cargo/core/compiler/build_context/target_info.rs#L682-L690
+                    let mut rf = String::with_capacity(val.len());
+                    if !val.is_empty() {
+                        for arg in val.split(' ').filter_map(|s| {
+                            let s = s.trim();
+                            if !s.is_empty() {
+                                Some(s)
+                            } else {
+                                None
+                            }
+                        }) {
```

Comment:
I'd `.map(|s| s.trim()).filter(|s| !s.is_empty())` this :wink:

Reply:
Heh I hadn't even opened the source link in `cargo` to see that they were doing the same, except for the `|s| s.trim()` -> `str::trim` :ok_hand:. This isn't only more concise, it's also closer to the reference implementation yet again.